wgsl/index.bs

```diff
@@ -90,7 +90,90 @@ that run on the GPU.
  * Features and semantics are exactly the ones of SPIR-V
  * Each item in this spec *must* provide the mapping to SPIR-V for the construct
 
-## Technical Overview TODO ## {#technical-overview}
+## Technical Overview ## {#technical-overview}
+
+WebGPU issues a unit of work to the GPU in the form of a <dfn noexport>provoking command</dfn>.
+[SHORTNAME] is concerned with two kinds of provoking commands:
+* a <dfn noexport>draw command</dfn> executes a [=GPURenderPipeline|render pipeline=]
+    in the context of [=pipeline input|inputs=], [=pipeline output|outputs=], and attached [=resources=].
+* a <dfn noexport>dispatch command</dfn> executes a [=GPUComputePipeline|compute pipeline=]
+    in the context of [=pipeline input|inputs=], [=pipeline output|outputs=], and attached [=resources=].
+
+Both kinds of pipelines use shaders written in [SHORTNAME].
+
+A <dfn noexport>shader</dfn> is the portion of a [SHORTNAME] program that executes a [=shader stage=] in a pipeline.
+A shader comprises:
+* An [=entry point=] [=function/function=].
+* The transitive closure of all called functions, starting with the entry point.
+    This set includes both [=user-defined function|user-defined=] and [=built-in function|built-in=] functions.
+    (For a more rigorous definition, see "[=functions in a shader stage=]".)
+* The set of variables and constants [=statically accessed=] by all those functions.
+* The set of types used to define or analyze all those functions, variables, and constants.
+
+When executing a shader stage, the implementation:
+* Binds [=resources=] to variables in the shader's [=resource interface of a shader|resource interface=],
+    making the contents of those resources available to the shader during execution.
+* Populates the formal parameters of the entry point, if they exist, with the stage's pipeline inputs.
+* Connects the entry point return value, if one exists, to the stage's pipeline outputs.
+* Then it invokes the entry point.
+
+A [SHORTNAME] program is organized into:
+* Functions, which specify execution behaviour.
+* Statements, which are declarations or units of executable behaviour.
+* Literals, which are text representations for pure mathematical values.
+* Constants, each providing a name for a value computed at a specific time.
+* Variables, each providing a name for storage for holding a value.
+* Function calls, each of which invokes a function to perform work, possibly producing a result value.
```

Comment:
Why are these separate? A function call is either a statement or an expression.

Reply:
I thought they were a big enough concept to call out, but can remove it.  Yes it's redundant.